cmd/event-notifier.go

```diff
@@ -537,6 +537,28 @@ func loadAllBucketNotifications(objAPI ObjectLayer) (map[string]*notificationCon
 	return nConfigs, lConfigs, nil
 }
 
+// addQueueTarget - calls newTargetFunc function and adds its returned value to queueTargets
+func addQueueTarget(queueTargets map[string]*logrus.Logger,
+	accountID, queueType string,
+	newTargetFunc func(string) (*logrus.Logger, error)) (string, error) {
+
+	// Construct the queue ARN for AMQP.
+	queueARN := minioSqs + serverConfig.GetRegion() + ":" + accountID + ":" + queueType
+
+	// Queue target if already initialized we move to the next ARN.
+	if _, ok := queueTargets[queueARN]; ok {
+		return queueARN, nil
+	}
+
+	// Using accountID we can now initialize a new AMQP logrus instance.
+	logger, err := newTargetFunc(accountID)
+	if err == nil {
+		queueTargets[queueARN] = logger
```

Comment:
This would make significant changes in current implementation.  Currently `Set()` and `Get()` equivalent code is used only in this function and no `Del()` anywhere.